### PR TITLE
Initialize JAVA_MAJOR_VERSION variable if missing

### LINKED ISSUE #92: JAVA_DIAGNOSTICS does not work out of the box
I made a little POC, in the shell `docker run -it openjdk sh`

```sh
# curl https://raw.githubusercontent.com/fabric8io-images/run-java-sh/master/fish-pepper/run-java-sh/fp-files/run-java.sh > run-java.sh
# chmod +x run-java.sh
# java `JAVA_DIAGNOSTICS=1 ./run-java.sh options` -version
```

The last command crashes

```text
Unrecognized VM option 'PrintGCDateStamps'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

To workaround to this issue I have to set manually the variable `JAVA_MAJOR_VERSION`.

I propose to consider to autodetect the right version using:
* the global `JAVA_VERSION` variable
* the file `$JAVA_HOME/release`
* the first line of `java -version` command

Same issue and solution for the AdoptOpenJDK image

### PR DESCRIPTION
**Pull request by fvasco**
Fix #92

I am not an experienced sh developer, so please carefully review this pr.

Tested with Java 7, 8, 11 ,13

### CODE REVIEW

**Comment by fvasco** [comment]
I checked CirleCI error, it is related to [this line](https://github.com/fabric8io-images/run-java-sh/blob/master/fish-pepper/run-java-sh/fp-files/run-java.sh#L434) on JVM 11: ParallelGC is not enabled on that version (but it works also on Java 13).

**Comment by rhuss** [comment]
Thanks ! Not sure whether I'm able to review it this week (as I'm on the road), but please ping me early next week in case I should forget ;)

File: fish-pepper/run-java-sh/fp-files/run-java.sh [hunk 1]
```diff
@@ -188,6 +188,24 @@ init_limit_env_vars() {
   fi
 }
 
+init_java_major_version() {
+    # Initialize JAVA_MAJOR_VERSION variable if missing
+    if [ -z "${JAVA_MAJOR_VERSION:-}" ]; then
+        # Parse JAVA_VERSION variable available in containers
+        if [ -n "${JAVA_VERSION:-}" ]; then
+            export JAVA_MAJOR_VERSION="`echo $JAVA_VERSION | sed -E 's/([0-9]+).*/\1/'`"
```
Comment (rhuss) [inline fish-pepper/run-java-sh/fp-files/run-java.sh]: `sed -E` is not known by every sed, it's better to use just `-e` which should be good enough for a simple regexp like this.

File: fish-pepper/run-java-sh/fp-files/run-java.sh [hunk 2]
```diff
@@ -188,6 +188,24 @@ init_limit_env_vars() {
...
+            return
+        fi
+        # Parse release file (translate "1.8" to "8")
+        if [ -r "${JAVA_HOME}/release" ]; then
+            export JAVA_MAJOR_VERSION="`grep -e '^JAVA_VERSION=' ${JAVA_HOME}/release | sed -E 's/.*\"(1\.)?([0-9]+).*/\2/'`"
```
Comment (rhuss) [inline fish-pepper/run-java-sh/fp-files/run-java.sh]: `-E` -> `-e`

File: fish-pepper/run-java-sh/fp-files/run-java.sh [hunk 3]
```diff
@@ -188,6 +188,24 @@ init_limit_env_vars() {
...
+            return
+        fi
+        # Parse java version output (translate "1.8" to "8")
+        export JAVA_MAJOR_VERSION="`java -version 2>&1 | head -1 | sed -E 's/.*\"(1\.)?([0-9]+).*/\2/'`"
```
Comment (rhuss) [inline fish-pepper/run-java-sh/fp-files/run-java.sh]: `-E` -> `-e`

**Comment by rhuss** [comment]
Ah, yes and it would be cool, if a bit of docs could be added to `fish-pepper/run-java-sh/fp-files/readme.md` could be added, too.

**Comment by fvasco** [comment]
@rhuss Any update?

**Comment by rhuss** [comment]
@fvasco sorry for the delay, will have a look tomorrow.

**Comment by rhuss** [comment]
Thanks ! Fixed the test and merged your contrib in another PR.

**Pull request closed without merging**